### PR TITLE
log4j-over-slf4j to slf4j migration

### PR DESCRIPTION
**Pull request by dsKarthick**
Two reasons we should do this
      1. Its best to use `sfl4j` directly rather than use `log4j-over-slf4j`
      2. `slf4j` provides better log event formatting than log4j

### CODE REVIEW

**Comment by dsKarthick** [comment]
@erikdw @brndnmtthws Please :eyes:

File: storm/src/main/storm/mesos/MesosSupervisor.java [hunk 1]
```diff
@@ -143,11 +146,11 @@ public void waitUntilRegistered() throws InterruptedException {
 
     @Override
     public void registered(ExecutorDriver driver, ExecutorInfo executorInfo, FrameworkInfo frameworkInfo, SlaveInfo slaveInfo) {
-      LOG.info("Received executor data <" + executorInfo.getData().toStringUtf8() + ">");
+      LOG.info("Received executor data < {} >", executorInfo.getData().toStringUtf8());
```
Comment (erikdw) [inline storm/src/main/storm/mesos/MesosSupervisor.java]: minor: this changes from `<foobar>` to `< foobar >`.   Probably don't care.

File: storm/src/main/storm/mesos/MesosSupervisor.java [hunk 2]
```diff
@@ -18,17 +18,22 @@
...
-import backtype.storm.generated.JavaObjectArg;
 import backtype.storm.scheduler.ISupervisor;
 import backtype.storm.utils.Utils;
-import org.apache.log4j.Logger;
+import clojure.lang.PersistentVector;
 import org.apache.mesos.Executor;
 import org.apache.mesos.ExecutorDriver;
 import org.apache.mesos.MesosExecutorDriver;
-import org.apache.mesos.Protos.*;
+import org.apache.mesos.Protos.ExecutorInfo;
```
Comment (erikdw) [inline storm/src/main/storm/mesos/MesosSupervisor.java]: this is great, death to star-imports!

File: storm/src/main/storm/mesos/MesosNimbus.java [hunk 3]
```diff
@@ -415,9 +449,11 @@ public boolean isHostAccepted(String hostname) {
       Collection<SupervisorDetails> existingSupervisors, Topologies topologies, Set<String> topologiesMissingAssignments) {
     synchronized (_offersLock) {
       LOG.debug("allSlotsAvailableForScheduling: Currently have " + _offers.size() + " offers buffered" +
```
Comment (erikdw) [inline storm/src/main/storm/mesos/MesosNimbus.java]: think you missed removing this line

File: storm/src/main/storm/mesos/MesosNimbus.java [hunk 4]
```diff
@@ -232,21 +265,22 @@ public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offers) {
...
         if (isHostAccepted(offer.getHostname())) {
-          LOG.debug("resourceOffers: Recording offer from host: " +
-              offer.getHostname() + ", offerId: " + offer.getId().getValue());
+          LOG.debug("resourceOffers: Recording offer from host: {}, offerId: {}",
+                    offer.getHostname(), offer.getId().getValue());
           _offers.put(offer.getId(), offer);
         } else {
-          LOG.debug("resourceOffers: Declining offer from host: " +
-              offer.getHostname() + ", offerId: " + offer.getId().getValue());
+          LOG.debug("resourceOffers: Declining offer from host: {} offerId: {}",
```
Comment (erikdw) [inline storm/src/main/storm/mesos/MesosNimbus.java]: minor: `,` has been removed, and is inconsistent with previous debug message for "Recording offer".  i.e., either remove comma from both, or leave in both.

**Comment by erikdw** [comment]
other than minor points this is awesome, thanks!

File: storm/src/main/storm/mesos/MesosNimbus.java (hunk 3 above)
Comment (dsKarthick) [inline storm/src/main/storm/mesos/MesosNimbus.java]: Fixed

File: storm/src/main/storm/mesos/MesosNimbus.java (hunk 4 above)
Comment (dsKarthick) [inline storm/src/main/storm/mesos/MesosNimbus.java]: Good catch. Fixed

File: storm/src/main/storm/mesos/MesosSupervisor.java (hunk 1 above)
Comment (dsKarthick) [inline storm/src/main/storm/mesos/MesosSupervisor.java]: Fixed

**Comment by erikdw** [comment]
LGTM @dsKarthick !  Thanks for making these changes.

**Comment by JessicaLHartog** [comment]
LGTM too! :+1:

**Comment by dsKarthick** [comment]
I squashed the commits on my local branch, cherry-picked it on to master and pushed it directly. Please see a49e8ce

**Pull request closed without merging**